frontend/src/app/app.component.html

```diff
@@ -73,59 +75,59 @@
           </mat-card-content>
         </mat-card>
         <div class="nav-buttons">
-          <button color="secondary" mat-raised-button matStepperPrevious>Předchozí krok</button>
-          <!-- TODO: Validace, ze je vse zaskrtnute ANO/NE -->
-          <button
-            color="primary"
-            [disabled]="basicInfoForm.invalid"
-            mat-raised-button matStepperNext
-          >Další krok</button>
+          <button color="secondary" mat-raised-button matStepperPrevious>
+            Předchozí krok
+          </button>
+          <button [disabled]="!canProceedToStep3"
+                  color="primary" mat-raised-button matStepperNext>
+            Další krok
+          </button>
         </div>
       </div>
     </mat-step>
     <mat-step label="Odeslání požadavku">
       <div class="container container--wide">
         <div class="confirmation-step">
-          <mat-card class="py-0">
+          <mat-card *ngIf="basicInfoForm" class="py-0">
             <mat-card-content>
               <div class="form-field">
                 <mat-label>Jméno:</mat-label>
-                <mat-label *ngIf="basicInfoForm.controls['firstName'].valid else unspecified;">
+                <mat-label *ngIf="basicInfoForm.controls['firstName']?.valid else unspecified;">
```

Comment:
jakto, ze to muze byt null?

Reply:
v případě, že control `firstName` neexistuje v tom formu